lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -191,6 +166,47 @@ def self.is_firebasecmd_supported?(cmd)
 
         true
       end
+
+      def self.get_token
+        client = Signet::OAuth2::Client.new(
+          token_credential_uri: 'https://oauth2.googleapis.com/token',
+          client_id: FirebaseAppDistributionLoginAction::CLIENT_ID,
+          client_secret: FirebaseAppDistributionLoginAction::CLIENT_SECRET,
+          refresh_token: ENV["FIREBASE_TOKEN"]
+        )
+
+        client.fetch_access_token!
+        return client.access_token
+      rescue Signet::AuthorizationError => error
+        UI.crash!("Wrong value for FIREBASE_TOKEN")
+      end
+
+      def self.get_app(params)
+        token = get_token
+
+        # begin
+        appId = (params[:app]).to_s
+
+        connection = Faraday.new(url: BASE_URL) do |conn|
+          conn.response(:json, parser_options: { symbolize_names: true })
+          conn.response(:raise_error) # raise_error middleware will run before the json middleware
+          conn.adapter(Faraday.default_adapter)
+        end
+
+        response = connection.get("#{PATH}#{appId}") do |request|
+          request.headers["Authorization"] = "Bearer " + token
+        end
+
+        contactEmail = response.body[:contactEmail]
+
+        if contactEmail.strip.empty?
+          UI.error("We could not find a contact email for app #{appId}. Please visit App Distribution within the Firebase Console to set one up.")
```

Comment:
I've updated in the errors used in the comment above, but here is some guidance on what errors to use:
https://docs.fastlane.tools/advanced/actions/#user-input-and-output

Based on that, I think an empty email and resource not found errors should used `UI.user_error!` and any other network error should use `UI.crash!`.